core/pta/device.c

```diff
@@ -21,18 +21,20 @@
 #define PTA_NAME "device.pta"
 
 static void add_ta(uint32_t flags, const TEE_UUID *uuid, uint8_t *buf,
-		   uint32_t blen, uint32_t *pos)
+		   uint32_t blen, uint32_t *pos, uint32_t rflags)
 {
-	if (flags & TA_FLAG_DEVICE_ENUM) {
-		if (*pos + sizeof(*uuid) <= blen)
-			tee_uuid_to_octets(buf + *pos, uuid);
+	if (!(flags & rflags))
```

Comment:
Please use positive logic also, like in the original code.
Avoiding to change lines that doesn't need to be changed is usually a good idea.